pkg/k8s/apis/cilium.io/v2/types.go

```diff
@@ -172,3 +173,50 @@ type CiliumNetworkPolicyList struct {
 	// Items is a list of CiliumNetworkPolicy
 	Items []CiliumNetworkPolicy `json:"items"`
 }
+
+// +genclient
+// +genclient:noStatus
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// CiliumEndpoint is the status of a Cilium policy rule
+// +k8s:openapi-gen=false
+type CiliumEndpoint struct {
+	// +k8s:openapi-gen=false
+	metav1.TypeMeta `json:",inline"`
+	// +k8s:openapi-gen=false
+	metav1.ObjectMeta `json:"metadata"`
+
+	Status CiliumEndpointDetail `json:"status"`
+}
+
+// CiliumEndpointDetail is the status of a Cilium policy rule
+// +k8s:deepcopy-gen=false
+type CiliumEndpointDetail models.Endpoint
+
+// DeepCopyInto is an inefficient hack to allow reusing models.Endpoint in the
+// CiliumEndpoint CRD.
+func (in *CiliumEndpointDetail) DeepCopyInto(out *CiliumEndpointDetail) {
```

Comment:
Does it hurt if we have the deepcopy generator in the `models` package to avoid this workaround? (I think it's doable)

Reply:
I think I can do that. I had left it here because DeepCopyInto is a k8s specific generated deepcopy, and i didn't want to confuse the types in models.

Reply:
Oh, no, they use the same thing. I'll change it to autogen in the other one.

Reply:
Hmm. Or not. I will look at this again later today. In the meantime, I'll push up my other fixups for review.